src/Nethermind/Nethermind.JsonRpc.Connectors/Client/BasicJsonRpcClient.cs

```diff
@@ -105,5 +109,23 @@ private void AddAuthorizationHeader()
 
         private static string Base64Encode(string plainText)
             => Convert.ToBase64String(Encoding.UTF8.GetBytes(plainText));
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    _client?.Dispose();
+                }
+                disposedValue = true;
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
```

Comment:
No finalizer no need for anything like that just do 
```suggestion
        protected virtual void Dispose()
        {
           _client.Dispose();        
        }
```